### DISCUSSION TITLE
Allowing insert for anon users?

### DISCUSSION THREAD

**Discussion by mrdbourke** [supabase/supabase#6757] [Questions] [answered]
Hello,

I'm planning on allowing auth later on.

But for now I'd like to enable anon users to write to the database.

I've turned on RLS and created a custom policy:

```
CREATE POLICY "Enable insert for all anon users" 
ON public.is_this_correct 
FOR INSERT WITH CHECK 
(auth.role() = 'anon');
```

However it still seems to be erroring.

Is this possible?

In essence, I'd like anyone to be able to write to the database but *not* be able to delete/update rows/tables.

**Comment by mrdbourke** [comment, marked as answer]
Actually, it seems it is as long as you put `returning: 'minimal'` to come back from Supabase (after an insert).

For example:

```javascript
let { data, error } = await supabase
    .from("is_this_correct")
    .insert([
        {
            id: ...,
            other_data: ...
        }],
       {
            returning: 'minimal' // add this to prevent errors when inserting with RLS on
 })
```

I got this from another comment on the Supabase GitHub discussions: https://github.com/supabase/supabase/discussions/6069#discussioncomment-2438430

**Reply by ukpagrace** [reply]
this is throwing an error

**Reply by alenjdev** [reply]
Based on the comment above the actual issue is there's no rule set for the `select` statement at the end of the function call, so you can either remove the `select` or create a rls policy that allows select for `anon` users as well

**Reply by ukpagrace** [reply]
yes, that was how I ended up solving it. Thanks
